### PR TITLE
hndlbrs to js

### PR DESCRIPTION
**Pull request by maximpristupa**

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -3,87 +3,75 @@
...
+    cb(null, 'uploads')
+  },
+  filename: function(req, file, cb) {
+    cb(null, file.fieldname + '-' + Date.now())
+  }
 })
 
 let upload = multer({
-    storage: storage
+  storage: storage
```
Comment (YanaYaroshevich) [inline server/server.js]: enough to have just 
{
   storage
}

File: server/views/script.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
+upload('newfile');
+
+function upload(file) {
+  if (file === 'newfile') {
+    const emptyCsvArray = [
```
Comment (YanaYaroshevich) [inline server/views/script.js]: o_o
can't you generate this one?

File: server/views/scss_style.scss [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  position: sticky;
+  left: 0vh;
+  background-color: $background-color; 
+  text-align: center;
+  padding-top: 2vh;
+}
+
+.number{
+  width: 2vh;
+  padding-bottom: 10px;
```
Comment (YanaYaroshevich) [inline server/views/scss_style.scss]: use more variables for sizes
they are reused several times

File: server/views/scss_style.scss [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  position: sticky;
+}
+
+.letter{
+  background-color: $background-color;
+  width: 150px;
+  text-align: center;
+}
+
+.numberRow{
```
Comment (YanaYaroshevich) [inline server/views/scss_style.scss]: use syntax .my-class-lala

Comment (YanaYaroshevich) [reply server/views/scss_style.scss]: with dashes i mean

File: server/views/index.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      <link rel="stylesheet" type="text/css" href="style.css">
+  </head>
+  <body>
+    <div class="container" onclick="closeCover()"></div>
+    <div class="head">
+        <form name="uploadFile" class="uploadFile">
+          <input type="file" id="getFile" name="myFile" onchange="upload(this.files[0])" class="myFileInput">
+        </form>
+
+        <button class="newFile" onclick="upload('newfile')">New File</button>
```
Comment (YanaYaroshevich) [inline server/views/index.html]: new-file

File: server/views/index.html [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        <button class="newFile" onclick="download()">Save File</button>
+        <button class="newFile" onclick="showModal()">Open Files</button>
+        <input class="cellInputForm" onkeydown = "if (event.keyCode == 13) writeCell()">
```
Comment (YanaYaroshevich) [inline server/views/index.html]: move if to function and pass event to it

File: server/views/index.html [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    </div>  
+
+    <div class="filename">
+      <p>File name:</p>
+      <input class="inputfilename" onkeydown = "if (event.keyCode == 13) writeFile()">
```
Comment (YanaYaroshevich) [inline server/views/index.html]: same

File: server/views/script.js [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+}
+
+function printHtml(html) {
+  document.getElementById('table').innerHTML = html;
+}
+
+  function calculate(filename) {
+    let formData = new FormData(document.forms.extable);
+    if (filename) {
+      formData.append("filename", filename);
```
Comment (YanaYaroshevich) [inline server/views/script.js]: single quotes, man

File: server/views/script.js [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+
+function changeCell(cell) {
+  input.oninput = function() {
+    cell.value = input.value;
+    highlightTheMentionedCell(input.value);
+  };
+}
+
+function writeCell() {
+  let cellId = input.getAttribute('cellId');
```
Comment (YanaYaroshevich) [inline server/views/script.js]: u use cellId a lot of times
move to const
same for other many times used string values

File: server/views/script.js [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+function writeInput(argument) {
+  let cellId = input.getAttribute('cellId');
+  let cell = document.getElementById(cellId);
+  input.value = cell.value;
+}
+
+function backlight(id, color) {
+  let numberId = id.match(/(\d+)_/)[1];
+  let letterId = id.match(/_(\d+)/)[1];
+  let numberCell = document.getElementById(numberId + '_number');
```
Comment (YanaYaroshevich) [inline server/views/script.js]: string template

File: server/views/script.js [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+}
+
+function writeInput(argument) {
+  let cellId = input.getAttribute('cellId');
+  let cell = document.getElementById(cellId);
+  input.value = cell.value;
+}
+
+function backlight(id, color) {
+  let numberId = id.match(/(\d+)_/)[1];
```
Comment (YanaYaroshevich) [inline server/views/script.js]: remember difference between let and const?

File: server/views/scss_style.scss [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+@mixin inputsize() {
+  text-align: center;
+  font-size: 30px;
+}
+
+body{
+  background-color: $background-color;
+}
+.head{
```
Comment (YanaYaroshevich) [inline server/views/scss_style.scss]: use inheritance 
(one selector in another) 
it should be formed as a tree

File: server/views/scss_style.scss [hunk 13]
```diff
@@ -0,0 +1,163 @@
+$cell-color: #fdfdfd;
+$background-color: #f3f3f5;
+
+@mixin inputsize() {
+  text-align: center;
+  font-size: 30px;
```
Comment (YanaYaroshevich) [inline server/views/scss_style.scss]: use rems for font

File: server/views/script.js [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+    cell.value = input.value;
+    highlightTheMentionedCell(input.value);
+  };
+}
+
+function writeCell() {
+  let cellId = input.getAttribute('cellId');
+  let cell = document.getElementById(cellId);
+  cell.value = input.value;
+  calculate();
```
Comment (YanaYaroshevich) [inline server/views/script.js]: just pass writeInput to calculate and evaluate it inside

File: server/views/script.js [hunk 15]
```diff
@@ -0,0 +1,224 @@
...
+  addClass('container', 'cover');
+}
+
+function showModal(argument) {
+  addClass('modal', 'showmodal');
+  addClass('container', 'cover');
+  let xhr = new XMLHttpRequest();
+  xhr.open("GET", "/lastfiles");
+  xhr.send();
+  xhr.onload = () => printFiles(xhr.response);
```
Comment (YanaYaroshevich) [inline server/views/script.js]: use right lambda syntax
if you don't return anything from the function, use curly brackets

File: server/server.js [hunk 16]
```diff
@@ -3,87 +3,75 @@
...
-        cb(null, 'uploads')
-    },
-    filename: function(req, file, cb) {
-        cb(null, file.fieldname + '-' + Date.now())
-    }
+  destination: function(req, file, cb) {
+    cb(null, 'uploads')
+  },
+  filename: function(req, file, cb) {
+    cb(null, file.fieldname + '-' + Date.now())
```
Comment (YanaYaroshevich) [inline server/server.js]: replace all string concats with string templates

File: server/views/script.js [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+  let modal = document.getElementsByClassName('modal')[0];
+  html = '';
+  if (json === '{}') {
+    html += '<p class="chooseFile">No files to open</p>';  
+    modal.innerHTML = html;
+    return;
+  }
+  html += '<p class="chooseFile">Choose File</p>';
+  let records = JSON.parse(json);
+  for (let k in records) {
```
Comment (YanaYaroshevich) [inline server/views/script.js]: don't use for-in
use built-in array methods

File: server/views/script.js [hunk 18]
```diff
@@ -0,0 +1,224 @@
...
+    const letter = links[i].match(/[A-Z]/)[0];
+    const number = links[i].match(/(\d+)/)[0];
+    let id = `${number - 1}_${letter.charCodeAt(0) - 65}`;
+    let linkCell = document.getElementById(id);
+    linkCell.classList.add("highlightRed");
+  }
+}
+
+function changeCell(cell) {
+  input.oninput = function() {
```
Comment (YanaYaroshevich) [inline server/views/script.js]: lambdas instead of anonymous functions

File: server/server.js (hunk 1 above)
Comment (maximpristupa) [reply server/server.js]: +

File: server/views/script.js (hunk 10 above)
Comment (maximpristupa) [reply server/views/script.js]: +

File: server/views/script.js (hunk 18 above)
Comment (maximpristupa) [reply server/views/script.js]: +

File: server/server.js (hunk 16 above)
Comment (maximpristupa) [reply server/server.js]: +

File: server/views/script.js (hunk 15 above)
Comment (maximpristupa) [reply server/views/script.js]: +

File: server/views/script.js (hunk 2 above)
Comment (maximpristupa) [reply server/views/script.js]: done

File: server/views/scss_style.scss (hunk 3 above)
Comment (maximpristupa) [reply server/views/scss_style.scss]: done

File: server/views/scss_style.scss (hunk 4 above)
Comment (maximpristupa) [reply server/views/scss_style.scss]: done

File: server/views/index.html (hunk 5 above)
Comment (maximpristupa) [reply server/views/index.html]: done

File: server/views/index.html (hunk 6 above)
Comment (maximpristupa) [reply server/views/index.html]: done

File: server/views/index.html (hunk 7 above)
Comment (maximpristupa) [reply server/views/index.html]: done

File: server/views/script.js (hunk 8 above)
Comment (maximpristupa) [reply server/views/script.js]: done

File: server/views/script.js (hunk 9 above)
Comment (maximpristupa) [reply server/views/script.js]: done

File: server/views/script.js (hunk 11 above)
Comment (maximpristupa) [reply server/views/script.js]: done

File: server/views/scss_style.scss (hunk 12 above)
Comment (maximpristupa) [reply server/views/scss_style.scss]: done

File: server/views/scss_style.scss (hunk 13 above)
Comment (maximpristupa) [reply server/views/scss_style.scss]: done

File: server/views/script.js (hunk 17 above)
Comment (maximpristupa) [reply server/views/script.js]: done

File: server/rw.js [hunk 19]
```diff
@@ -25,8 +25,10 @@ function objectsToCsv(objects) {
                 row.push('')
             }
         }
+        while (row.length < 20) { row.push(''); }
```
Comment (YanaYaroshevich) [inline server/rw.js]: magic numbers

File: server/server.js [hunk 20]
```diff
@@ -3,87 +3,75 @@
...
-    }
-    let objects = sort.csvToObjects(subarray);
-    let calculatedCsv = rw.objectsToCsv(objects);
-    res.send(displayPage(calculatedCsv));
-    // res.render("calc", {csv: calculatedCsv, rowCount: calculatedCsv.length});
-})
-
-app.listen(3000, () => console.log('Server started on port 3000'));
+app.post('/calculate', upload.single('extable'), (req, res, next) => {
+  let csv = req.body.cell;
```
Comment (YanaYaroshevich) [inline server/server.js]: let and const

File: server/server.js [hunk 21]
```diff
@@ -3,87 +3,75 @@
...
+  let rowCount = req.body.rowCount;
+  let size = csv.length / +rowCount;
+  let subarray = [];
+  for (let i = 0; i < Math.ceil(csv.length / size); i++) {
+    subarray[i] = csv.slice((i * size), (i * size) + size).join();
+  }
+  let objects = sort.csvToObjects(subarray);
+  let calculatedCsv = rw.objectsToCsv(objects);
+  let file = 'file' + Date.now() + '.csv';
```
Comment (YanaYaroshevich) [inline server/server.js]: string template

File: server/views/script.js [hunk 22]
```diff
@@ -0,0 +1,209 @@
...
+    formData.append('filename', filename);
+  }
+  let xhr = new XMLHttpRequest();
+  xhr.open('POST', '/calculate');
+  xhr.send(formData);
+
+  xhr.onload = () => {
+    printHtml(displayPage(xhr.response));
+  }
+  if (callback) {
```
Comment (YanaYaroshevich) [inline server/views/script.js]: shouldn't it be done in onload?

File: server/views/script.js [hunk 23]
```diff
@@ -0,0 +1,209 @@
+upload('newfile');
+
+function upload(file) {
+  if (file === 'newfile') {
+    const emptyCsvArray = Array(30).fill().map(() => Array(17).fill().map(() => ''));
```
Comment (YanaYaroshevich) [inline server/views/script.js]: what is 30 and 17? please do consts

File: server/rw.js (hunk 19 above)
Comment (maximpristupa) [reply server/rw.js]: done

File: server/server.js (hunk 20 above)
Comment (maximpristupa) [reply server/server.js]: done

File: server/server.js (hunk 21 above)
Comment (maximpristupa) [reply server/server.js]: done

File: server/views/script.js (hunk 22 above)
Comment (maximpristupa) [reply server/views/script.js]: done

File: server/views/script.js (hunk 23 above)
Comment (maximpristupa) [reply server/views/script.js]: done

**Pull request merged**